### DISCUSSION TITLE
Модели

### DISCUSSION THREAD

**Discussion by ghost** [ArtemA1ekseev/telegram-bot-animal-shelter#2] [Q&A] [answered]
Нужны ли нам классы cat и dog при наличии класса Pet? Я просто вчера их написала, но сегодня увидела у Ильнура класс Pet c такими же полями?

**Comment by qwsq4** [comment]
Я думаю классы cat и dog сделать наследниками класса pet

**Comment by ArtemA1ekseev** [comment, marked as answer]
Попробую ответить на этот вопрос.
Изначально я представлял себе такие модели:

```java
Cat {
 @Id
    @GeneratedValue
    private Long id;
    private String breed;
    private String name;
    private int yearOfBirth;
    private String description;
}
```

```java
Dog {
    @Id
    @GeneratedValue
    private Long id;
    private String breed;
    private String name;
    private int yearOfBirth;
    private String description;
}
```
```java
ReportData {
    @Id
    @GeneratedValue
    private long id;

    private Long chatId;

    private String ration;

    private String health;

    private String habits;

    private long days;

    private String filePath;

    private long fileSize;

    @Lob
    private byte[] data;

    private String caption;

    private Date lastMessage;

    private Long lastMessageMs;

    public ReportData() {

    }
```

Как ранее мне про статус написал Ильнур (@ilnurGatin), я могу предложить такой вариант:

```java
public enum Status {

    APPROVED,
    REFUSED,
    TRIAL_PERIOD,
    SEARCH
}
```

Написать этот код можно прям внутри класса, это уже встречалась в прошлых проектах.
Еще 2 класса у нас могут быть, классами для пользователей, которые выбирают животного.
**PersonCat + PersonDog** 
Например вот так.
Но я все еще не исключаю вариант, при котором мы просто сделаем родительский класс, и будем работать с ним. На данный момент проекта, можно даже оставить так, как вы предлагаете, но не стоит удивляться, что потребуется изменение в бедующем.
